files/image_config/platform/rc.local

```diff
@@ -240,16 +251,26 @@ fi
 
 program_console_speed
 
+platform=""
+if [ -n "$aboot_platform" ]; then
+    platform=$aboot_platform
+elif [ -n "$onie_platform" ]; then
+    platform=$onie_platform
+else
+    echo "Unknown SONiC platform"
+fi
+
+# W/A for https://github.com/sonic-net/sonic-buildimage/issues/24086
+if /usr/bin/jq -e 'has("DPU")' /usr/share/sonic/device/$platform/platform.json 2>/dev/null; then
+    echo "DPU type detected. Requesting IP ..."
+    request_dpu_midplane_ip
```

Comment:
Will there be any retry if this fails?

Reply:
No, eth0-midplane will already be created by this point. 

This problem is only seen when the dhcp_server is started and we do image upgrade. So, dhcp_server is expected to be present at this point